### PR TITLE
Switching z3 refutation on by default.

### PR DESCRIPTION
**Pull request by dkrupp**

### CODE REVIEW

File: analyzer/codechecker_analyzer/cmd/analyze.py [hunk 1]
```diff
@@ -303,10 +303,11 @@ def add_arguments_to_parser(parser):
...
 
         analyzer_opts.add_argument('--z3-refutation',
-                                   action='store_true',
                                    dest='enable_z3_refutation',
-                                   default=argparse.SUPPRESS,
-                                   help="Enable the Z3 SMT Solver backend to "
+                                   choices=['on', 'off'],
+                                   default='on',
+                                   help="Switch on/off the Z3 SMT Solver "
+                                        "backend to "
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/cmd/analyze.py]: Please update the user guide: https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#analyzer-configuration-

File: analyzer/codechecker_analyzer/cmd/analyze.py [hunk 2]
```diff
@@ -303,10 +303,11 @@ def add_arguments_to_parser(parser):
                                         "constraint solver.")
 
         analyzer_opts.add_argument('--z3-refutation',
-                                   action='store_true',
                                    dest='enable_z3_refutation',
-                                   default=argparse.SUPPRESS,
-                                   help="Enable the Z3 SMT Solver backend to "
+                                   choices=['on', 'off'],
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/cmd/analyze.py]: Do not we want to do the same with the `--z3` option?

File: analyzer/codechecker_analyzer/cmd/check.py [hunk 3]
```diff
@@ -332,10 +332,11 @@ def add_arguments_to_parser(parser):
...
 
         analyzer_opts.add_argument('--z3-refutation',
-                                   action='store_true',
                                    dest='enable_z3_refutation',
-                                   default=argparse.SUPPRESS,
-                                   help="Enable the Z3 SMT Solver backend to "
+                                   choices=['on', 'off'],
+                                   default='on',
+                                   help="Switch on/off the Z3 SMT Solver "
+                                        "backend to "
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/cmd/check.py]: Please update the user guide too: https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#check-

**Comment by gyorb** [comment]
the feature was finished in #2222

**Pull request closed without merging**